tests/pybaseball/datahelpers/caching/test_caching.py

```diff
@@ -0,0 +1,344 @@
+import os
+import pathlib
+import pickle
+import shutil
+from datetime import datetime, timedelta
+from typing import Any, Dict, List, Tuple
+from unittest.mock import MagicMock, mock_open, patch
+
+import pandas as pd
+import pytest
+from _pytest.monkeypatch import MonkeyPatch
+
+from pybaseball.datahelpers import caching
+
+@pytest.fixture(name='save_mock')
+def _save_mock(monkeypatch: MonkeyPatch) -> MagicMock:
+    mock = MagicMock()
+    monkeypatch.setattr(caching.dataframe_cache, 'save', mock)
+    return mock
+
+
+@pytest.fixture(name="mock_data_1")
+def _mock_data_1() -> pd.DataFrame:
+    return pd.DataFrame([1, 2], columns=['a'])
+
+
+@pytest.fixture(name="mock_data_2")
+def _mock_data_2() -> pd.DataFrame:
+    return pd.DataFrame([[1, 2], [3, 4]], columns=['a', 'b'])
+
+
+class TestCacheWrapper:
```

Comment:
Can this go on a diet? This is really long and I don't know what it does (which is a recurring problem across our unit testing honestly)

Reply:
Its really long hecause it tests the entire cache thoroughly. So no it cannot go on a diet unless we want to have untested code scenarios, which I can't even imagine a compelling reason for.